### PR TITLE
Improved dark color mode support

### PR DESCRIPTION
**Pull request by cpsievert**
Bootstrap 5.3 (which we updated to in https://github.com/rstudio/bslib/pull/749) introduced the concept of [color modes](https://getbootstrap.com/docs/5.3/customize/color-modes/#dark-mode) and also adds a "1st class" dark mode. This PR updates some of bslib's Sass code to be more compatible with dark mode.

These changes alone aren't sufficient for the best dark experience. You'll also need https://github.com/rstudio/shiny/pull/3882 and https://github.com/rstudio/shiny/pull/3875 for Shiny-based styling to work well.

This PR also adds an (internal, for now) `bslib:::input_dark_mode_switch()` to make it easier to get a dark mode toggle into demo apps, but we'll wait to add an official exported option until a later PR.

### CODE REVIEW

File: inst/builtin/bs5/shiny/ionrangeslider/_variables.scss [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 
 $line_height: 1px !default;
 $line_bg: none !default;
-$line_bg_color: $gray-600 !default;
+$line_bg_color: RGBA($emphasis-color-rgb, 0.65) !default;
 $line_border: none !default;
 
 $handle_color: $component-active-bg !default;
-$handle_color_hover: tint-color($handle_color, 15%) !default;
+$handle_color_hover: rgba($handle_color, 0.85) !default;
```
Comment (cpsievert) [inline inst/builtin/bs5/shiny/ionrangeslider/_variables.scss:15]:
```suggestion
$handle_color_hover: RGBA($handle_color, 0.85) !default;
```

File: R/input-switch.R [hunk 2]
```diff
@@ -73,6 +73,17 @@ toggle_switch <- function(id, value = NULL, session = get_current_session()) {
   session$onFlush(callback, once = TRUE)
 }
 
+# Keep this internal for now until we agree on the UX
+input_dark_mode_switch <- function(id, label, ..., width = "auto") {
+  res <- input_switch(id, label, ..., width = width)
+  # TODO: .navbar should probably do this for .shiny-input-container
```
Comment (gadenbuie) [inline R/input-switch.R:79]: or would we add `.navbar` to the set of gap-spaced elements?

Comment (cpsievert) [reply R/input-switch.R:79]: I guess that's an option but also a bit weird considering it's not `flex-direction: column`

**Review by gadenbuie — APPROVED** [review]
This all looks great, can't wait for 🌞 🌚 !

**Pull request merged**